<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>kmemo</class>
 <widget class="QMainWindow" name="kmemo">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>K-memo - 智能任务管理器</string>
  </property>
  <property name="styleSheet">
   <string>QMainWindow {
    background: #eff6ff;
    font-family: "Segoe UI", Arial, sans-serif;
}

/* 卡片通用样式 */
.CardFrame {
    background: white;
    border: none;
    border-radius: 12px;
    margin: 8px;
}

/* 按钮样式 */
QPushButton {
    border: none;
    border-radius: 8px;
    font-weight: 500;
}

/* 图标按钮样式 */
.IconButton {
    background: transparent;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 18px;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}

.IconButton:hover {
    background: rgba(66, 133, 244, 0.1);
}

/* 蓝色主色调按钮 */
.PrimaryButton {
    background: #4285f4;
    color: white;
    padding: 8px 16px;
    font-size: 14px;
}

.PrimaryButton:hover {
    background: #1976d2;
}

/* 绿色确认按钮 */
.SuccessButton {
    background: #34a853;
    color: white;
    padding: 8px 16px;
    font-size: 14px;
}

.SuccessButton:hover {
    background: #137333;
}

/* 红色取消按钮 */
.DangerButton {
    background: #ea4335;
    color: white;
    padding: 8px 16px;
    font-size: 14px;
}

.DangerButton:hover {
    background: #d33b2c;
}

/* 任务项样式 */
QListWidget {
    background: transparent;
    border: none;
    outline: none;
}

QListWidget::item {
    background: transparent;
    border: none;
    padding: 8px 0px;
    margin: 0px;
}

/* 输入框样式 */
QTextEdit {
    background: white;
    border: 1px solid #dadce0;
    border-radius: 8px;
    padding: 12px;
    font-size: 14px;
    outline: none;
}

QTextEdit:focus {
    border-color: #4285f4;
    background: white;
}

/* 复选框样式 */
QCheckBox {
    font-size: 14px;
    color: #3c4043;
}

QCheckBox::indicator {
    width: 20px;
    height: 20px;
    border: 2px solid #dadce0;
    border-radius: 4px;
    background: white;
}

QCheckBox::indicator:checked {
    background: #4285f4;
    border-color: #4285f4;
    image: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTEzLjUgNEw2IDExLjVMMi41IDgiIHN0cm9rZT0id2hpdGUiIHN0cm9rZS13aWR0aD0iMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIi8+Cjwvc3ZnPgo=);
}
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainVerticalLayout">
    <property name="spacing">
     <number>16</number>
    </property>
    <property name="leftMargin">
     <number>8</number>
    </property>
    <property name="topMargin">
     <number>32</number>
    </property>
    <property name="rightMargin">
     <number>8</number>
    </property>
    <property name="bottomMargin">
     <number>32</number>
    </property>

    <!-- 顶部统计卡片 -->
    <item>
     <widget class="QFrame" name="statsCard">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="styleSheet">
       <string>QFrame#statsCard {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.05);
    border-radius: 12px;
    padding: 16px;
}
       </string>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>80</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="statsLayout">
       <property name="leftMargin">
        <number>16</number>
       </property>
       <property name="topMargin">
        <number>16</number>
       </property>
       <property name="rightMargin">
        <number>16</number>
       </property>
       <property name="bottomMargin">
        <number>16</number>
       </property>
       <item>
        <widget class="QLabel" name="statsIconLabel">
         <property name="text">
          <string>📅</string>
         </property>
         <property name="styleSheet">
          <string>font-size: 24px; color: #4285f4;</string>
         </property>
         <property name="maximumSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="taskCountLabel">
         <property name="text">
          <string>2个待办事项</string>
         </property>
         <property name="styleSheet">
          <string>font-size: 16px; font-weight: 500; color: #3c4043; margin-left: 8px;</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="statsHeaderSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="advancedFeaturesBtn">
         <property name="text">
          <string>👥</string>
         </property>
         <property name="toolTip">
          <string>高级功能</string>
         </property>
         <property name="styleSheet">
          <string>QPushButton {
    background: transparent;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 18px;
    color: #4285f4;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}
QPushButton:hover {
    background: rgba(66, 133, 244, 0.1);
}
          </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="menuBtn">
         <property name="text">
          <string>☰</string>
         </property>
         <property name="toolTip">
          <string>菜单</string>
         </property>
         <property name="styleSheet">
          <string>QPushButton {
    background: transparent;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 18px;
    color: #5f6368;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}
QPushButton:hover {
    background: rgba(95, 99, 104, 0.1);
}
          </string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- 中间添加任务卡片 -->
    <item>
     <widget class="QFrame" name="addTaskCard">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="styleSheet">
       <string>QFrame#addTaskCard {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.05);
    border-radius: 12px;
    padding: 16px;
}
       </string>
      </property>
      <layout class="QVBoxLayout" name="addTaskLayout">
       <property name="leftMargin">
        <number>16</number>
       </property>
       <property name="topMargin">
        <number>16</number>
       </property>
       <property name="rightMargin">
        <number>16</number>
       </property>
       <property name="bottomMargin">
        <number>16</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="addTaskHeaderLayout">
         <item>
          <widget class="QLabel" name="addTaskLabel">
           <property name="text">
            <string>添加新待办</string>
           </property>
           <property name="styleSheet">
            <string>font-size: 16px; font-weight: 500; color: #3c4043;</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="addTaskHeaderSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="confirmAddBtn">
           <property name="text">
            <string>✓</string>
           </property>
           <property name="toolTip">
            <string>确认添加</string>
           </property>
           <property name="styleSheet">
            <string>QPushButton {
    background: #34a853;
    color: white;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 16px;
    font-weight: bold;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}
QPushButton:hover {
    background: #137333;
}
            </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="cancelAddBtn">
           <property name="text">
            <string>✗</string>
           </property>
           <property name="toolTip">
            <string>取消</string>
           </property>
           <property name="styleSheet">
            <string>QPushButton {
    background: #ea4335;
    color: white;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 16px;
    font-weight: bold;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}
QPushButton:hover {
    background: #d33b2c;
}
            </string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="addTaskHintLabel">
         <property name="text">
          <string>请输入待办内容</string>
         </property>
         <property name="styleSheet">
          <string>font-size: 12px; color: #5f6368; margin: 4px 0px;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="taskInputTextEdit">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>80</height>
          </size>
         </property>
         <property name="placeholderText">
          <string>写作业，在写完之后上传到班级群，把昨天剩下...</string>
         </property>
         <property name="styleSheet">
          <string>QTextEdit {
    background: white;
    border: 1px solid #dadce0;
    border-radius: 8px;
    padding: 12px;
    font-size: 14px;
    outline: none;
}
QTextEdit:focus {
    border-color: #4285f4;
    background: white;
}
          </string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- 底部任务列表卡片 -->
    <item>
     <widget class="QFrame" name="taskListCard">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="styleSheet">
       <string>QFrame#taskListCard {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.05);
    border-radius: 12px;
    padding: 16px;
}
       </string>
      </property>
      <layout class="QVBoxLayout" name="taskListLayout">
       <property name="leftMargin">
        <number>16</number>
       </property>
       <property name="topMargin">
        <number>16</number>
       </property>
       <property name="rightMargin">
        <number>16</number>
       </property>
       <property name="bottomMargin">
        <number>16</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="taskListHeaderLayout">
         <item>
          <widget class="QLabel" name="taskListLabel">
           <property name="text">
            <string>全部待办</string>
           </property>
           <property name="styleSheet">
            <string>font-size: 16px; font-weight: 500; color: #3c4043;</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="taskListHeaderSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="sortTasksBtn">
           <property name="text">
            <string>⇅</string>
           </property>
           <property name="toolTip">
            <string>排序</string>
           </property>
           <property name="styleSheet">
            <string>QPushButton {
    background: transparent;
    border: none;
    border-radius: 20px;
    padding: 8px;
    font-size: 18px;
    color: #4285f4;
    min-width: 40px;
    max-width: 40px;
    min-height: 40px;
    max-height: 40px;
}
QPushButton:hover {
    background: rgba(66, 133, 244, 0.1);
}
            </string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QListWidget" name="taskListWidget">
         <property name="styleSheet">
          <string>QListWidget {
    background: transparent;
    border: none;
    outline: none;
}

QListWidget::item {
    background: transparent;
    border: none;
    padding: 8px 0px;
    margin: 0px;
}
          </string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>

  <!-- 菜单栏 -->
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>400</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="advancedMenu">
    <property name="title">
     <string>高级功能</string>
    </property>
    <addaction name="actionDetailedManagement"/>
    <addaction name="actionStatisticsAnalysis"/>
    <addaction name="actionQuickPanel"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <addaction name="advancedMenu"/>
  </widget>

  <!-- 动作定义 -->
  <action name="actionDetailedManagement">
   <property name="text">
    <string>详细任务管理</string>
   </property>
   <property name="toolTip">
    <string>打开详细任务管理面板</string>
   </property>
  </action>
  <action name="actionStatisticsAnalysis">
   <property name="text">
    <string>统计分析</string>
   </property>
   <property name="toolTip">
    <string>查看任务统计和分析</string>
   </property>
  </action>
  <action name="actionQuickPanel">
   <property name="text">
    <string>快速操作面板</string>
   </property>
   <property name="toolTip">
    <string>打开快速操作面板</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>设置</string>
   </property>
   <property name="toolTip">
    <string>应用程序设置</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
  </action>
  <action name="actionAddTask">
   <property name="text">
    <string>新建任务</string>
   </property>
   <property name="toolTip">
    <string>创建新任务</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>